app/backend/src/tests/test_activeness_probes.py

```diff
@@ -0,0 +1,175 @@
+        hosting_status=HostingStatus.can_host,
+        meetup_status=MeetupStatus.wants_to_meetup,
+        last_active=now() - timedelta(days=335),
+    )
+
+    with session_scope() as session:
+        queue_job(session, "send_activeness_probes", empty_pb2.Empty())
+
+    process_jobs()
+
+    with real_jail_session(token) as jail:
+        res = jail.JailInfo(empty_pb2.Empty())
+        assert res.has_pending_activeness_probe
+        assert res.jailed
+
+        res = jail.RespondToActivenessProbe(
+            jail_pb2.RespondToActivenessProbeReq(response=jail_pb2.ACTIVENESS_PROBE_RESPONSE_STILL_ACTIVE)
+        )
+        assert not res.has_pending_activeness_probe
+        assert not res.jailed
+
+    with api_session(token) as api:
+        res = api.GetUser(api_pb2.GetUserReq(user=user.username))
+        assert res.hosting_status == api_pb2.HOSTING_STATUS_CAN_HOST
+        assert res.meetup_status == api_pb2.MEETUP_STATUS_WANTS_TO_MEETUP
+
+    push_collector.assert_user_has_single_matching(
+        user.id,
+        title="Are you still hosting on Couchers.org?",
+        body="Please log back in and let us know if you are still open to hosting!",
+    )
+
+
+def test_activeness_probes_disabled(db, push_collector):
+    new_config = config.copy()
+    new_config["ACTIVENESS_PROBES_ENABLED"] = False
+
+    with patch("couchers.jobs.handlers.config", new_config):
+        user, token = generate_user(
+            hosting_status=HostingStatus.can_host,
+            meetup_status=MeetupStatus.wants_to_meetup,
+            last_active=now() - timedelta(days=335),
+        )
+
+        with session_scope() as session:
+            queue_job(session, "send_activeness_probes", empty_pb2.Empty())
+
+        process_jobs()
+
+        with real_jail_session(token) as jail:
+            res = jail.JailInfo(empty_pb2.Empty())
+            assert not res.has_pending_activeness_probe
+            assert not res.jailed
+
+        with session_scope() as session:
+            assert not session.execute(select(ActivenessProbe)).scalar_one_or_none()
+
+
+def test_activeness_probes_expiry(db, push_collector):
+    user, token = generate_user(
+        hosting_status=HostingStatus.can_host,
+        meetup_status=MeetupStatus.wants_to_meetup,
+        last_active=now() - timedelta(days=335),
+    )
+
+    with session_scope() as session:
+        queue_job(session, "send_activeness_probes", empty_pb2.Empty())
+
+    process_jobs()
+
+    with real_jail_session(token) as jail:
+        res = jail.JailInfo(empty_pb2.Empty())
+        assert res.has_pending_activeness_probe
+        assert res.jailed
+
+    with session_scope() as session:
+        probe = session.execute(select(ActivenessProbe)).scalar_one()
+        probe.probe_initiated = now() - timedelta(days=15)
+        assert probe.notifications_sent == 1
+        probe.notifications_sent = 2
+
+        queue_job(session, "send_activeness_probes", empty_pb2.Empty())
```

Comment:
Where is the timeline for responding to the activeness probe before their status gets changed? I see it in these tests, but I don't see it in the non-test code?

I also wonder if we should send one more reminder? But we can do that later.